priv/repo/migrations/20160531152824_add_package_extra_index.exs

```diff
@@ -0,0 +1,14 @@
+defmodule HexWeb.Repo.Migrations.AddPackageExtraIndex do
+  use Ecto.Migration
+
+  def up do
+    execute """
+      CREATE INDEX packages_extra_text ON
+        packages USING GIN (to_tsvector('english', regexp_replace((meta->'extra')::text, '/', ' ')))
+    """
```

Comment:
This will not work if you want to specify paths so we can't use full text search unfortunately. This code is encoding the json as text so you will have a string with "{...}".

If we wan't full text search we need to concatenate all text inside the nested json values. I think exact value search would be better.